sdk/textanalytics/Azure.AI.TextAnalytics/src/CategorizedEntity.cs

```diff
@@ -13,16 +13,20 @@ namespace Azure.AI.TextAnalytics
     /// </summary>
     public readonly struct CategorizedEntity
     {
-        internal CategorizedEntity(Entity entity)
+        internal CategorizedEntity(Entity entity) : this(entity.Text, entity.Category, entity.Subcategory, entity.ConfidenceScore, entity.Offset, entity.Length)
+        {
+        }
+
+        internal CategorizedEntity(string text, string category, string subcategory, double confidenceScore, int offset, int length)
```

Comment:
I intentionally avoided using the legacy type directly in constructors.  My goal was to not allow the legacy surface to get deeply embedded so that it can be removed at a later point, as there are potentially plans to sunset the associated REST service.